api/base/filters.py

```diff
@@ -66,6 +69,33 @@ def filter_queryset(self, request, queryset, view):
             return queryset.sort(*ordering)
         return queryset
 
+    def check_serializer_fields(self, queryset, query_params, view, ordering):
+        serializer_class = view.serializer_class
+        if not queryset:
+            return []
+        sorting_params = []
+        for i, field in enumerate(query_params.getlist(self.ordering_param)):
+            preserve_order = ''
+            # If the field is already a source, it will be returned in the ordering list
+            if field in ordering:
+                sorting_params.append(field)
+                continue
+            # The field may have a '-' before the name for descending order. This needs to be preserved
+            if field[0] == '-':
+                preserve_order = '-'
+                field = field[1:]
+            if field in serializer_class._declared_fields:
+                # Checking if the field can be sorted on
+                source_field = serializer_class._declared_fields[field]
+                source_field_name = source_field.source
+                # Checking if the original sort param could be used to sort
+                if getattr(queryset[0], field, 'None') != 'None':
```

Comment:
Is this tested for fields that actually equal the string None? I'm confused as to why you've stringified this. I think you just want a try/catch here.

Reply:
or
```suggestion
if getattr(queryset[0], field, None):
```